spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/service/impl/DefaultTaskExecutionService.java

```diff
@@ -267,7 +273,13 @@ public String getLog(String platformName, String taskId) {
 				if (taskDeployment == null) {
 					throw new IllegalArgumentException();
 				}
-				taskId = taskDeployment.getTaskDefinitionName();
+				String taskName = taskDeployment.getTaskDefinitionName();
+				TaskExecution taskExecution = this.taskExplorer.getLatestTaskExecutionForTaskName(taskName);
+				if (taskExecution != null && !taskExecution.getExternalExecutionId().equals(taskId)) {
+					return "";
```

Comment:
Should we have same type of logic of throwing an error instead of returning empty log?

Reply:
This is mostly driven by the Data Flow UI. If the returned log value is empty, then we don't display anything, if there is anything else (including the exception message), that will get displayed at the logs page.

Reply:
I was just thinking to show a reason why we don't show anything. It's a difference whether log is empty or if log is missing.

Reply:
We wanted to display the log only for the latest task execution in case of Cloud Foundry and having this check on the platform type and the latest task execution on the data flow UI side turns out to be cumbersome. Hence, this PR returns empty log for anything other than latest task execution on CF and thereby handling at the backend. Happy to choose any other approach that can help achieve this better.